In [1]:
%load_ext autoreload
%autoreload

import sys
import time

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

from tabulate import tabulate
from joblib import dump, load
from sklearn import preprocessing
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import GridSearchCV, RandomizedSearchCV
from sklearn.feature_selection import SelectFromModel
from sklearn.preprocessing import StandardScaler, Normalizer
from sklearn.decomposition import PCA
from sklearn.metrics import roc_curve, auc

sys.path.append('../')
from utils import feature_selection, thomas_parser, PCA_reduction, perf, time_comparison
from toBoolean import convert

### PCA

In [2]:
PCA_reduction('../../dumps/2020.03.11-17.39.csv','tree')

  Variance    Training acc    Test acc    Components    Time (s)
----------  --------------  ----------  ------------  ----------
      1           0.950859    0.946957           119    0.616202
      0.99        0.950859    0.946957            98    0.498316
      0.95        0.950859    0.946957            77    0.401685
      0.9         0.950859    0.946957            60    0.334009
      0.85        0.950859    0.946957            47    0.288169


In this case the accuracies are 100% similar but we managed to reduced the number of features from 119 to 47, so more than 60%. The time as also be improved by half !